### ISSUE TITLE
Ref parameter - cannot use reference to this

### ISSUE THREAD

**Issue by LeonidVeriga** [bridgedotnet/Bridge#1569]
### Steps To Reproduce
https://dotnetfiddle.net/7qwXlu
https://deck.net/b60b6107c8a83cb84a324da0f8e5389b
``` csharp
public class App
{
    public static void Main()
    {
        new S().RefThisTest();
    }

  private struct S
   {
       public int i, j;

       public S(int i, int j)
       {
           this.i = i;
           this.j = j;
       }

       public void RefThisTest()
       {
           i = 11;
           j = 12;
           M(ref this);
           Assert.AreEqual(i, 42);
           Assert.AreEqual(j, 43);
       }

       private static void M(ref S s)
       {
           Assert.AreEqual(s.i, 11);
           Assert.AreEqual(s.j, 12);
           s = new S(42, 43);
           Assert.AreEqual(s.i, 42);
           Assert.AreEqual(s.j, 43);
       }
   }
}
```

**Comment by michaelcheers** [comment]
@LeonidVeriga, it's not allowed in native C#, [see this](https://dotnetfiddle.net/gEGhVn).

**Comment by LeonidVeriga** [comment]
Hi @michaelcheers, the code sample in the steps to reproduce allowed- [.Net Fiddle](https://dotnetfiddle.net/BTQfOV)

**Comment by LeonidVeriga** [comment]
Fixed by PR #2994
